docs/wslvar.1
.TH "WSLVAR" "1" "DATEPLACEHOLDER" "VERSIONPLACEHOLDER" "WSL Utilities User Manual"
.SH NAME
.B wslvar
- Component of WSL Utilities
.SH SYNOPSIS
.B wslvar
.RB [ \-sl ]
.I NAME
.PP
.B wslvar
.RB [\-hvSL ]
.SH DESCRIPTION
This is a WSL tool to help you get Windows system environment variables.
.SH OPTIONS
.SS "Output Type Options"
.B -s, --sys
use data from system local & global variables.
.TP
.B -l, --shell
use data from Shell folder environment variables.
.SS "Other Options"
.TP
.B -h, --help
print a simple help.
.TP
.B -v, --version
print current version.
.TP
.B -S, --getsys
show available system local & global variables.
.TP
.B -L, --getshell
show available Shell folder environment variables.
.SH CONFIGURATION
.TP
.B WSLVAR_DEFAULT_VARTYPE
 Default Type of variable to get when not specifying option. Default type is 1. Here are the available Types:
.nf
.in +4
1: User Shell Folders
2: Windows Environment Variables
.in
.fi
.SH AUTHOR
Created by Patrick Wu <me@patrickwu.space>
.SH REPORTING BUGS
Report bugs to <https://github.com/wslutilities/wslu/issues>;
For Ubuntu specific/related bugs, report to <https://bugs.launchpad.net/ubuntu/+source/wslu>.
.SH COPYRIGHT
This is free software; you can redistribute it and/or modify it under
the terms of the GNU GPL version 3 or (at your option) any later
version.
There is NO warranty; not even MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
.SH SEE ALSO
wslu(7)

debian/changelog
wslu (3.2.3-0kali1) kali-dev; urgency=medium

  * wslsys: fix an issue in call_windows_uptime
    (GH: #182)
  * header: fix a distro detection issue in 
    Ubuntu Impish (21.10)
  * debian/control: fix dependency issue in 
    installing wslu (LP: #1926771)
  * docs: include available commands in manpage 
    (GH: #169)
  * project: fix a problem that Powershell output
    non-unicode character (GH: #162)
  * wslview: fix an issue that caused the relative
    Linux path unable to be opened (GH: #165)
  * wslusc: fix an issue that some people unable
    to open their GUI application (GH: #174)
  * header: Avoid error message when the file doesn't
    exist (LP: 1911379)

 -- Patrick Wu <me@patrickwu.space>  Mon, 14 Jun 2021 22:14:49 +0800

wslu (3.2.2-0kali1) kali-dev; urgency=medium

  * docs: include available commands in manpage 
    (GH: #169)
  * project: fix a problem that Powershell output
    non-unicode character (GH: #162)
  * wslview: fix an issue that caused the relative
    Linux path unable to be opened (GH: #165)
  * wslusc: fix an issue that some people unable
    to open their GUI application (GH: #174)

 -- Patrick Wu <me@patrickwu.space>  Mon, 15 Mar 2021 10:52:33 +0800

wslu (3.2.1-0kali1) kali-dev; urgency=medium

  * header: improvements on the system drive detection. (closes #133)
  * tests: update the test system and tests. (closes #125)
  * wslact: add back-compatibility for the auto-mounting feature.
  * wslview: fix an issue when handling http/https
  * wslview: fix a bug that it won't open file:///mnt/c/... or /mnt/c/...
  * wslusc: fix an issue that wsl.exe execute with wrongly passed param (close #138)
  * wslusc: fix the problem that the previous version that GUI shortcut generated won't work.
  * wslsys: fix an issue that the WSL Version shown is wrong.
  * Includes a wslview.desktop (PR #143 and #146)
  * Update the script to fix badly generated shortcut issue (fixes #141)
  * Fix a DPI script related space issue (fixes #144)

 -- Patrick Wu <me@patrickwu.space>  Wed, 23 Dec 2020 22:33:07 +0800

wslu (3.1.1-0kali1) kali-dev; urgency=medium

  * project: add separate Fedora Remix support
  * header: fix `ubuntu.exe` detection issue
  * wslview: mistake in calling a variable (fixes #118 again)
  * wslusc: always use the absolute path (fixes #129)
  * wslusc: fixes a possible white space path issue
  * wslusc: When TMP is empty, try TEMP

 -- Patrick Wu <me@patrickwu.space>  Wed, 01 Jul 2020 22:22:53 +0800

wslu (3.1.0-0kali1) kali-dev; urgency=medium

  * header: use the proper color definition
  * wslfetch: new display for WSL with WSL version displayed
  * wslfetch: use the proper logo for Ubuntu (Reported by @flexiondotorg)
  * wslsys: add WSL version option
  * wslsys: fix an issue in DPI scaling getting wrong value due to the API change in Latest fast ring insider build (Reported by @crramirez)
  * wslact: adding `--mount-options` for auto-mounting feature
  * wslact: fix an i18n related issue in time-sync feature (fixes #113 by @ricardosantos9521)
  * wslgsu: removed temporarily
  * wslgsu: limit the Windows version it can use
  * wslview: properly detect all different protocols and path styles (fixes #118)
  * package: fix alpine dependency issue (fixes #117)

 -- Patrick Wu <me@patrickwu.space>  Wed, 13 May 2020 22:53:03 +0800

wslu (3.0.1-0kali1) kali-dev; urgency=medium

  * Initial release

 -- Patrick Wu <me@patrickwu.space>  Mon, 13 Apr 2020 20:41:40 +0800


src/wslvar.sh
# shellcheck shell=bash
var_type=1

help_short="wslvar [-sl] NAME\nwslvar [-hvSL]"

function call_shell {
	winps_exec "(Get-ItemProperty 'HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\User Shell Folders').'$*'" | cat
}

function view_shell {
	winps_exec "Get-ItemProperty 'HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\User Shell Folders'"| tail -n +3 | head -n -10
}

function call_sys {
	winps_exec "Write-Output \${Env:$*}" | cat
}

function view_sys {
	winps_exec "Get-ChildItem env:" | tail -n +2 | head -n -2
}

function cl_destoryer {
	echo "$@" | tr -d "\r"
}

function caller {
	if [ "$*" != "" ]; then
		case $WSLVAR_DEFAULT_VARTYPE in
			1) p="$(cl_destoryer "$(call_sys "$@")")";;
			2) p="$(cl_destoryer "$(call_shell "$@")")";;
			*) error_echo "Invalid variable type. Aborted." 22;;
		esac
		echo "$p"
	else
		error_echo "No Input. Aborted." 21
	fi
}

while [ "$1" != "" ]; do
	case "$1" in
		-s|--sys) WSLVAR_DEFAULT_VARTYPE=1; shift;;
		-l|--shell) WSLVAR_DEFAULT_VARTYPE=2; shift;;
		-S|--getsys) view_sys; exit;;
		-L|--getshell) view_shell; exit;;
		-h|--help) help "$0" "$help_short"; exit;;
		-v|--version) version; exit;;
		*) caller "$@"; exit;;
	esac
done

error_echo "No Input. Aborted." 21

